src/Helpers/IPFSClient.ts

```diff
@@ -0,0 +1,98 @@
+import * as FormData from 'form-data'
+import fetch, { Response } from 'node-fetch'
+import * as str from 'string-to-stream'
+
+interface IPFSConfiguration {
+  readonly ipfsUrl: string
+}
+
+interface FetchOptions {
+  readonly timeout: number
+}
+
+enum Type {
+  File = 'File',
+  Directory = 'Directory',
+}
+
+interface IPFSObject {
+  readonly Name?: string
+  readonly Hash: string
+  readonly Size?: number
+  readonly Type?: Type
+  readonly Links?: ReadonlyArray<IPFSObject>
+}
+
+interface LSResult {
+  Arguments: {
+    [key: string]: string
+  }
+  Objects: {
+    [key: string]: IPFSObject
+  }
+}
+
+type addFileToDirectory = (directoryHash: string, fileName: string, filehash: string) => Promise<IPFSObject>
+
+type addText = (text: string, x?: FetchOptions) => Promise<IPFSObject>
+
+type cat = (s: string, x?: FetchOptions) => Promise<Response['buffer']>
+
+type createEmptyDirectory = () => Promise<IPFSObject>
+
+type getDirectoryFileHashes = (s: string) => Promise<ReadonlyArray<string>>
+
+type ls = (s: string) => Promise<LSResult>
+
+export class IPFS {
+  private readonly url: string
+
+  constructor(configuration: IPFSConfiguration) {
+    this.url = configuration.ipfsUrl
+  }
+
+  addFileToDirectory: addFileToDirectory = async (directoryHash, fileName, fileHash) => {
+    const response = await fetch(
+      `${this.url}/api/v0/object/patch/add-link?arg=${directoryHash}&arg=${fileName}&arg=${fileHash}`
+    )
+    return await response.json()
+  }
+
+  addText: addText = async (text: string, fetchOptions) => {
```

Comment:
Is the type 'string' needed here given the definition on line 37?
btw, I am finding the use of 'type' makes the functions easier to read.

Reply:
Nope, good catch thanks! :D

Same here, I like that I can read through all the types and interfaces without the implementation details.